usecases/schema/tenant.go

```diff
@@ -260,24 +280,72 @@ func IsLocalActiveTenant(phys *sharding.Physical, localNode string) bool {
 		phys.Status == models.TenantActivityStatusHOT
 }
 
-func (h *Handler) getTenantsByNames(class string, names []string) ([]*models.Tenant, error) {
+func (h *Handler) getTenantsByNames(class string, names []string, after *string, limit *int64) ([]*models.Tenant, error) {
 	info, err := h.multiTenancy(class)
 	if err != nil || info.Tenants == 0 {
 		return nil, err
 	}
 
 	ts := make([]*models.Tenant, 0, len(names))
 	f := func(_ *models.Class, ss *sharding.State) error {
+		if after != nil && limit != nil {
+			ts = getAllTenantsPriorityQueue(ss.Physical, *after, int(*limit))
+			return nil
+		}
+
+		if limit != nil {
+			ts = make([]*models.Tenant, 0, *limit)
+		} else if len(names) == 0 {
+			ts = make([]*models.Tenant, len(ss.Physical))
+		} else {
+			ts = make([]*models.Tenant, 0, len(names))
+		}
+
+		i := 0
 		for _, name := range names {
 			if _, ok := ss.Physical[name]; !ok {
 				continue
 			}
-			ts = append(ts, &models.Tenant{
-				Name:           name,
-				ActivityStatus: schema.ActivityStatus(ss.Physical[name].Status),
-			})
+
+			ts[i] = makeTenant(name, entSchema.ActivityStatus(ss.Physical[name].Status))
+			if limit != nil && len(ts) >= int(*limit) {
+				break
+			}
+			i++
 		}
 		return nil
 	}
 	return ts, h.metaReader.Read(class, f)
 }
+
+func makeTenant(name, status string) *models.Tenant {
+	return &models.Tenant{
+		Name:           name,
+		ActivityStatus: status,
+	}
+}
+
+func getAllTenantsPriorityQueue(shards map[string]sharding.Physical, after string, limit int) []*models.Tenant {
```

Comment:
This looks to me like essentially (exactly?) the same function as the one in `cluster/store/schema.go`, can we avoid duplicating this logic?